src/core/compiler.jl

```diff
@@ -245,36 +266,29 @@ end
 ```
 """
 macro model(fexpr)
-
-    # translate all ~ occurences to macro calls
-    fexpr = translate(fexpr)
-
     # extract model name (:name), arguments (:args), (:kwargs) and definition (:body)
     modeldef = MacroTools.splitdef(fexpr)
-
     # function body of the model is empty
-    if all(l -> isa(l, LineNumberNode), modeldef[:body].args)
-        @warn("Model definition seems empty, still continue.")
-    end
-
+    warn_empty(modeldef[:body])
     # construct compiler dictionary
     compiler = Dict(
         :name => modeldef[:name],
         :closure_name => Symbol(modeldef[:name], :_model),
-        :args => [],
+        :args => modeldef[:args],
```

Comment:
Hmm, the whole compiler `Dict` will probably go away or get restructured. I wanted to start with something simple and I believe `compiler[:args]` was being used somewhere with the assumption that it is essentially `modeldef[:args]` so I did this. This may have been a source of errors, so I will double check it next weekend.